src/netbase.cpp

```diff
@@ -728,6 +724,13 @@ bool SetSocketNonBlocking(SOCKET& hSocket, bool fNonBlocking)
     return true;
 }
 
+bool SetSocketNoDelay(SOCKET& hSocket)
+{
+    int set = 1;
+    int rc = setsockopt(hSocket, IPPROTO_TCP, TCP_NODELAY, (const char*)&set, sizeof(int));
+    return rc != -1;
```

Comment:
I'd feel slightly better with `rc == 0`. 0 is guaranteed to be the success value both in Windows and UNIX, whereas on failure [windows returns](https://msdn.microsoft.com/en-us/library/windows/desktop/ms740476(v=vs.85).aspx) `SOCKET_ERROR` (which may or may not be -1, it's not mentioned).